src/core/components/files.js

```diff
@@ -87,6 +89,19 @@ function normalizeContent (opts, content) {
   })
 }
 
+function pinFile (self, opts, file, cb) {
+  // Pin a file if it is the root dir of a recursive add or the single file
+  // of a direct add.
+  const pin = 'pin' in opts ? opts.pin : true
+  const isRootDir = !file.path.includes('/')
+  const shouldPin = pin && isRootDir && !opts.onlyHash && !opts.hashAlg
```

Comment:
Yeah, this is related to https://github.com/ipfs/js-ipfs/issues/1373#event-1660703561. `hashAlg` creates an incorrect DAG and requires changes to IPLD before we can support it.

Reply:
If the `hashAlg` is the default value it would be ok? I think we need to call back with an error if `hashAlg` is passed, instead of silently not pinning.

Reply:
No, I don't think it'd work. IIRC, it's more like 'IPLD can't handle CIDv1' and anytime `hash` is provided there is a cast to v1 [here](https://github.com/ipfs/js-ipfs/blob/master/src/core/components/files.js#L31) (as it should and is documented in the command's help text).

Example: An object's `Hash` is CIDv0 while the key it was stored under is v1
![image](https://user-images.githubusercontent.com/2660142/41568208-fcc0d7a2-7329-11e8-9ec4-f35011d5b363.png)

I am happy to respond with an error though I must ask should we just revert the functionality?